lib/puma/launcher.rb

```diff
@@ -262,6 +262,17 @@ def prune_bundler
         "#{d.name}:#{spec.version.to_s}"
       end
 
+      if @options[:extra_runtime_dependencies]
+        @options[:extra_runtime_dependencies].each do |d_name|
+          spec = Bundler.rubygems.loaded_specs(d_name) rescue nil
```

Comment:
Rather than a `rescue nil`, we should do a proper begin/rescue here, which will get rid of the the `if/else`.

Reply:
There's actually no need for a rescue here at all, I don't know what I was thinking https://www.rubydoc.info/github/bundler/bundler/Bundler/RubygemsIntegration#loaded_specs-instance_method, `loaded_specs` is just a hash lookup. 🤦‍♂️